### PR TITLE
Add a test for device list updates of local users coming down /sync

### PR DESCRIPTION
**Pull request by anoadragon453**
This PR adds a test that: checks device list changes are propagated to other users on the local homeserver, if those users share a room.

This was created in part to help test https://github.com/matrix-org/synapse/pull/11905, until I realised that we're going to need to reach deeper into Synapse in order to do what we want.

Still, this is hopefully useful.

### CODE REVIEW

File: tests/csapi/device_lists_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	bob := deployment.Client(t, "hs1", "@bob:hs1")
+
+	// Deduce alice's device ID
+	resp := alice.MustDoFunc(
+		t,
+		"GET",
+		[]string{"_matrix", "client", "v3", "devices"},
+	)
+	responseBodyBytes := client.ParseJSON(t, resp)
+	aliceDeviceID := gjson.GetBytes(responseBodyBytes, "devices.0.device_id").Str
```
[the remark this answers is not in the document]
Comment (kegsay) [reply tests/csapi/device_lists_test.go:32]: ` /_matrix/client/v3/account/whoami` is probably slightly cleaner as this returns `device_id`.

**Review by kegsay — APPROVED** [review]
Use whoami and dendrite should be happier too.

**Comment by anoadragon453** [comment]
Have updated, however it looks like Dendrite doesn't support the v3 version of the `whoami` endpoint. Shall I change the test to use `r0` instead?

**Comment by kegsay** [comment]
No, I'll just add v3 to Dendrite and merge this when it has landed, assuming you're happy.

**Comment by kegsay** [comment]
This has taken a bit longer than planned because it's kinda annoying to actually do this using `gorilla/mux`. I'll get there..

**Comment by kegsay** [comment]
Tests still failing because `/whoami` isn't returning a device ID: `{"user_id":"@alice:hs1"}` on dendrite.

**Comment by kegsay** [comment]
Tests should pass now.

**Pull request merged**